pykickstart/commands/syspurpose.py

```diff
@@ -0,0 +1,82 @@
+#
+# Martin Kolman <mkolman@redhat.com>
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use, modify,
+# copy, or redistribute it subject to the terms and conditions of the GNU
+# General Public License v.2.  This program is distributed in the hope that it
+# will be useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  Any Red Hat
+# trademarks that are incorporated in the source code or documentation are not
+# subject to the GNU General Public License and may only be used or replicated
+# with the express permission of Red Hat, Inc.
+#
+
+from pykickstart.base import BaseData, KickstartCommand
+from pykickstart.errors import KickstartParseError
+from pykickstart.options import KSOptionParser
+from pykickstart.version import F29
+
+from pykickstart.i18n import _
+
+class F29_SystemPurpose(KickstartCommand):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        KickstartCommand.__init__(self, writePriority, *args, **kwargs)
+        self.op = self._getParser()
+
+        self.role = kwargs.get("role", None)
+        self.sla = kwargs.get("sla", None)
+        self.usage = kwargs.get("usage", None)
+
+    def __str__(self):
+        retval = KickstartCommand.__str__(self)
+
+        if not retval and not any([self.role, self.sla, self.usage]):
+            return ""
+
+        retval += "# Intended system purpose\nsyspurpose"
+        if self.role:
+            retval+=" --role=%s" % self.role
+        if self.sla:
+            retval+=" --sla=%s" % self.sla
+        if self.usage:
+            retval+=" --usage=%s" % self.usage
+
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="syspurpose", description="""
```

Comment:
The name `syspurpose` looks really weird. It might be better to use just `purpose` or we could start to support long command names, then it could be `system-purpose`.